File: netpulse-api/internal/incidents/attribution.go
package incidents

import (
	"strings"

	"github.com/shashank-4bt/NetPulse/netpulse-api/internal/contract"
)

func Title(evidenceClass, isolatedLayer, namedNetwork string) string {
	if evidenceClass == "measured_fact" && isolatedLayer == "isp" && strings.TrimSpace(namedNetwork) != "" {
		return "Measured failures isolated to network " + namedNetwork
	}
	if evidenceClass == "measured_fact" && isolatedLayer == "service" {
		return "Elevated connectivity failures observed toward the service"
	}
	return contract.ObservedFailures
}

func RejectsCausalOverclaim(title string) bool {
	lower := strings.ToLower(title)
	return strings.Contains(lower, "caused the outage") || strings.Contains(lower, "is down for everyone")
}

File: netpulse-web/src/features/observatory/attribution.ts
import { OBSERVED_FAILURES_TITLE } from "@/domain/observatory";

export type AttributionInput = {
  evidenceClass: string;
  isolatedLayer: string | null;
  namedNetwork: string | null;
};

export function incidentTitle(input: AttributionInput): string {
  if (
    input.evidenceClass === "measured_fact" &&
    input.isolatedLayer === "isp" &&
    input.namedNetwork
  ) {
    return `Measured failures isolated to network ${input.namedNetwork}`;
  }
  if (
    input.evidenceClass === "measured_fact" &&
    input.isolatedLayer === "service"
  ) {
    return "Elevated connectivity failures observed toward the service";
  }
  return OBSERVED_FAILURES_TITLE;
}

export function isCausalOverclaim(title: string): boolean {
  const lower = title.toLowerCase();
  return (
    lower.includes("caused the outage") ||
    lower.includes("is down for everyone")
  );
}
